### PR TITLE
Better reduction modulo n

### LINKED ISSUE #12592: Sympy can't simplify this modular expression
Sympy can simplify `(i*n + j%n) % n` into `j%n`. 
But `((j % n)*n + j%n) % n` can't be simplified.

### PR DESCRIPTION
**Pull request by enedil**
If some terms of `Add` are multiples of modulus, they are discarded.

Fixes #12592

### CODE REVIEW

File: sympy/core/mod.py [hunk 1]
```diff
@@ -97,11 +97,14 @@ def doit(p, q):
         if p.is_Add:
             args = []
             for i in p.args:
-                a = cls(i, q)
-                if a.count(cls) > i.count(cls):
-                    args.append(i)
+                if (i/q).is_integer:
+                    pass
```
Comment (jksuom) [inline sympy/core/mod.py]: This could probably be `continue`. Then the next `else` (and added indentation) would not be needed.

**Comment by enedil** [comment]
So I updated

File: sympy/core/mod.py [hunk 2]
```diff
@@ -97,6 +97,9 @@ def doit(p, q):
         if p.is_Add:
             args = []
             for i in p.args:
+                if (i/q).is_integer:
+                    continue
+
```
Comment (jksuom) [inline sympy/core/mod.py]: This line contains trailing whitespace. (Run `bin/test quality`.)

**Comment by enedil** [comment]
Sorry for the mess, I'm still learning how to use git. Now I fixed.

File: sympy/core/mod.py [hunk 3]
```diff
@@ -6,21 +6,16 @@
...
     """Represents a modulo operation on symbolic expressions.
-
     Receives two arguments, dividend p and divisor q.
-
     The convention used is the same as Python's: the remainder always has the
     same sign as the divisor.
-
     Examples
     ========
-
```
Comment (smichr) [inline sympy/core/mod.py]: there must be a blank like after the `===` or else it doesn't format properly

File: sympy/core/mod.py [hunk 4]
```diff
@@ -6,21 +6,16 @@
 
 class Mod(Function):
     """Represents a modulo operation on symbolic expressions.
-
```
Comment (smichr) [inline sympy/core/mod.py]: I prefer the whitespace here and below.

**Comment by smichr** [comment]
A test that fails in master but passes here should be added to the test suite.

**Comment by enedil** [comment]
@smichr how to check which tests don't fail? I just don't think those new tests that no longer fail are not related and thus shouldn't be corrected in this pull request. I don't have computational power to run all tests on all supported Python versions and Travis seems not much informant in that matter. Maybe I miss something.

**Comment by smichr** [comment]
Something like `assert  (i*n + j%n) % n  == j%n` like appears in the issue could be added to the appropriate file. See in core/tests which file has the Mod tests in it and include this expression there.

**Comment by gxyd** [comment]
@enedil can you please address the requested changes?

**Comment by gxyd** [comment]
This PR doesn't fix the issue as of now (I verified locally). Also no appropriate test has been added.

This PR has been inactive for quite sometime, so we are going to close it and open an issue to reference it. We try to keep our pull request queue small and focused on active work.  We encourage you to reopen the pull request if and when you continue to work on this. Please contact us if you have any questions.

Thanks for contributing.

see https://github.com/sympy/sympy/wiki/Closing-pull-requests for our policies on closing pull request.

We already have issue #12592 opened for it.

**Pull request closed without merging**